tests/e2e/unified_test.go

```diff
@@ -376,6 +376,31 @@ func runScenario(ctx context.Context, t *testing.T, testPause bool, fixture reso
 					}
 				}
 
+				if ShouldTestRereconiliation(t, primaryResource) {
+					eventsBefore := h.Events.HTTPEvents
+
+					touchObject(h, primaryResource)
+					// Pause to allow re-reconciliation
+					// (annotations don't change the generation, so we can't wait for observedGeneration)
+					time.Sleep(2 * time.Second)
+
+					eventsAfter := h.Events.HTTPEvents
+
+					h.Events.HTTPEvents = eventsBefore
+
+					for i := len(eventsBefore); i < len(eventsAfter); i++ {
+						event := eventsAfter[i]
+						isReadOnly := false
+						switch event.Request.Method {
+						case "GET":
+							isReadOnly = true
+						}
+						if !isReadOnly {
+							t.Errorf("FAIL: unexpected event during re-reconciliation: %v", event)
+						}
+					}
+				}
```

Comment:
Could be redundant, but can we also verify the generation is not updated in the object? I'm concerned a controller may write back to spec unexpectedly as each controller has its own logic.